source/includes/security/authentication.kt

```diff
@@ -0,0 +1,200 @@
+            )
+        }
+        .credential(credential)
+        .build()
+
+val mongoClient = MongoClient.create(settings)
+// end-scramsha1-mongo-cred
+
+// AWS Authentication
+
+// start-aws-sdk-mcred
+val credential = MongoCredential.createAwsCredential(null, null)
+
+val settings = MongoClientSettings.builder()
+    .applyToClusterSettings { builder: ClusterSettings.Builder ->
+        builder.hosts(
+            listOf(ServerAddress("<atlasUri>"))
+        )
+    }
+    .credential(credential)
+    .build()
+
+val mongoClient = MongoClient.create(settings)
+// end-aws-sdk-mcred
+
+// start-aws-sdk-cred-string
+val mongoClient =
+    MongoClient.create("mongodb://<atlasUri>?authMechanism=MONGODB-AWS")
+// end-aws-sdk-cred-string
+
+
+// start-aws-env-mcred
+val credential = MongoCredential.createAwsCredential(null, null)
+
+val settings = MongoClientSettings.builder()
+    .applyToClusterSettings { builder: ClusterSettings.Builder ->
+        builder.hosts(
+            listOf(ServerAddress("<atlasUri>"))
+        )
+    }
+    .credential(credential)
+    .build()
+
+val mongoClient = MongoClient.create(settings)
+// end-aws-env-mcred
+
+// start-aws-env-cred-string
+val mongoClient =
+    MongoClient.create("mongodb://<atlasUri>?authMechanism=MONGODB-AWS")
+// end-aws-env-cred-string
+
+// start-aws-mcred
+val credential = MongoCredential.createAwsCredential("<awsKeyId>", "<awsSecretKey>".toCharArray())
+
+val settings = MongoClientSettings.builder()
+        .applyToClusterSettings { builder: ClusterSettings.Builder ->
+            builder.hosts(
+                listOf(ServerAddress("<atlasUri>"))
+            )
+        }
+        .credential(credential)
+        .build()
+
+val mongoClient = MongoClient.create(settings)
+// end-aws-mcred
+
+// start-aws-mcred-wmechprop
+val credential = MongoCredential.createAwsCredential("<awsKeyId>", "<awsSecretKey>".toCharArray())
+    .withMechanismProperty("AWS_SESSION_TOKEN", "<awsSessionToken>")
+
+val settings = MongoClientSettings.builder()
+        .applyToClusterSettings { builder: ClusterSettings.Builder ->
+            builder.hosts(
+                listOf(ServerAddress("<atlasUri>"))
+            )
+        }
+        .credential(credential)
+        .build()
+
+val mongoClient = MongoClient.create(settings)
+// end-aws-mcred-wmechprop
+
+// start-aws-lambda-expression
+val awsFreshCredentialSupplier: Supplier<AwsCredential> = Supplier {
+    // Add your code here to fetch new credentials
+
+    // Return the new credentials
+    AwsCredential("<awsKeyId>", "<awsSecretKey>", "<awsSessionToken>")
+}
+
+val credential = MongoCredential.createAwsCredential("<awsKeyId>", "<awsSecretKey>".toCharArray())
+    .withMechanismProperty(MongoCredential.AWS_CREDENTIAL_PROVIDER_KEY, awsFreshCredentialSupplier)
+
+val settings = MongoClientSettings.builder()
+    .applyToClusterSettings { builder ->
+        builder.hosts(listOf(ServerAddress("<hostname>", <port>)))
```

Comment:
Port is an integer and is specified as <port> (without quotes) in the [Java documentation](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/auth/). To maintain consistency, I suggest we follow the same format:
```suggestion
                listOf(ServerAddress("<hostname>", <port>))
```

Reply:
Apologies, this one actually don't need any changes.